Week #3

Rafnes opened the pull request.

Rafnes committed:
add: добавлены колонки password и enabled в таблицу users
add: добавлены поля password и enabled в энтити User

Rafnes committed:
add: добавлены колонки password и enabled в таблицу users (#51)

add: добавлены колонки password и enabled в таблицу users
add: добавлены поля password и enabled в энтити User

gface34rus committed:
add: конструктор в AdsDTO, логика в методы AdServiceImpl

gface34rus committed:
add: List<Ad> findAllByAuthorId
fix: public AdsDTO getUserAds

Rafnes committed:
add: конструктор в AdsDTO, логика в методы AdServiceImpl (#52)

add: конструктор в AdsDTO, логика в методы AdServiceImpl

Rafnes committed:
add: добавлены классы для авторизации SecurityUser, UserDetailsServiceImpl
refactor: изменен WebSecurityConfig, добавлен UserDetailsServiceImpl

Fet2113 committed:
add @CrossOrigin в контроллеры Ad\Comment\User

Fet2113 committed:
add @CrossOrigin в контроллеры Ad,Comment,User (#54)

gface34rus committed:
add: логика в методы класса CommentServiceImpl
add: findAllByAdId(Integer adId) в CommentRepository

Rafnes committed:
[не работает, в процессе]
refactor: AuthController, AuthServiceImpl, UserDetailsServiceImpl и WebSecurityConfig изменен для регистрации с учетом добавления данных в репозиторий
delete: бин InMemoryUserDetailsManager из WebSecurityConfig

Rafnes committed:
fix: регистрация починена

ibogomolova committed:
add: Image, ImageRepository, AdNotFoundException, UserNotFoundException, methods in ImageForAdServiceImpl, changeSet in changelog.yml, OneToOne & Join

ibogomolova committed:
fix: the path to the image is saved in the DB

Fet2113 committed:
fix AdServiceImpl добавлена заглушка чтобы не падал фронт, refactor UserController добавлена аутентификация,  refactor UserService, add добавлена логика в методы класса UserServiceImpl, UserRepository

ibogomolova committed:
add: добавлены классы для авторизации SecurityUser, UserDetailsServiceImpl (#53)

add: добавлены классы для авторизации SecurityUser,
UserDetailsServiceImpl
refactor: изменен WebSecurityConfig, добавлен UserDetailsServiceImpl

ibogomolova committed:
add: логика в методы класса CommentServiceImpl add: findAllByAdId(Integer adId) в CommentRepository (#55)

add: логика в методы класса CommentServiceImpl
add: findAllByAdId(Integer adId) в CommentRepository

ibogomolova committed:
Merge branch 'dev' into bogomolova

ibogomolova committed:
add: Image (#56)

... ImageRepository, AdNotFoundException, UserNotFoundExcepti…on,
methods in ImageForAdServiceImpl, changeSet in changelog.yml, OneToOne &
Join

ibogomolova committed:
fix AdServiceImpl, refactor UserController,  refactor UserService, add добавлена логика в методы класса UserServiceImpl, UserRepository (#57)

fix AdServiceImpl добавлена заглушка чтобы не падал фронт, refactor
UserController добавлена аутентификация, refactor UserService, add
добавлена логика в методы класса UserServiceImpl, UserRepository

ibogomolova committed:
add @PreAuthorize in controllers

ibogomolova committed:
Merge remote-tracking branch 'origin/dev' into bogomolova

ibogomolova committed:
fix @PreAuthorize in CommentController

ibogomolova committed:
fix №2 @PreAuthorize in CommentController

Rafnes committed:
add @PreAuthorize in controllers (#68)

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -51,10 +51,16 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+    @Bean
+    public UserDetailsService userDetailsService(UserRepository userRepository) {
+        return new UserDetailsServiceImpl(userRepository);
+    }
```
У нас уже есть такой бин, не нужно создавать еще раз

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -51,10 +51,16 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                                         .authenticated())
                 .cors()
                 .and()
+                .userDetailsService(userDetailsService(userRepository))
```
он по умолчанию подтягивает бин, это не нужно

Rafnes replied:
удалено

Rafnes replied:
удалено

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/controller/AdController.java:
```diff
@@ -34,34 +43,39 @@ public ResponseEntity<AdDTO> addAd(@RequestPart("properties") CreateOrUpdateAdDT
     }
 
     @Operation(summary = "Получить объявление по ID", description = "Возвращает объявление по указанному ID")
+    @PreAuthorize(USER + " or " + ADMIN)
```
А это отличается от ALL?

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/model/Image.java:
```diff
@@ -0,0 +1,64 @@
+
+    @OneToOne
+    @JoinColumn(name = "Ad_id")
+    private Ad ad;
```
Я бы связь настраивал только со стороны юзера и рекламы

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/roles/Permissions.java:
```diff
@@ -0,0 +1,7 @@
+package ru.skypro.homework.roles;
+
+public interface Permissions {
+    String ADMIN = "hasRole('ADMIN')";
```
на встрече я показал это как класс Permissions но по сути это RoleAuthority. Я скорее это название использовал для примера

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/security/UserDetailsServiceImpl.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        log.info("Вошли в loadUserByUsername");
```
этот лог нам ничего не дает, можно только если  debug

Rafnes replied:
удалено вместе с аннотацией

Rafnes replied:
Удалено

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java:
```diff
@@ -1,47 +1,190 @@
     public AdsDTO getUserAds() {
-        return new AdsDTO();
+        UserDetails userDetails = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Integer userId = 1;
```
почему 1?

AleksandrZhukovJava replied:
почему 1? х2

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java:
```diff
@@ -1,47 +1,190 @@
     @Override
-    public void updateAdImage(Integer id, MultipartFile image) {
+    public void updateAdImage(Integer id, MultipartFile imageFile) {
+        Ad ad = adRepository.findById(id).orElseThrow();
```
кастомную ошибку

AleksandrZhukovJava replied:
кастомную ошибку х2

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/AuthServiceImpl.java:
```diff
@@ -1,46 +1,54 @@
-                        .build());
+        if (registerDTO.getPassword().length() < 8) {
+            throw new IllegalArgumentException("Пароль должен быть длиннее 8 символов");
+        }
```
эту валидацию можно сделать через @Size внутри самой дто

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/ImageForAdServiceImpl.java:
```diff
@@ -1,32 +1,147 @@
+        try {
+            resizedImage = resizeImage(file.getBytes(), 800, 600); // Устанавливаем размеры 800x600
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to resize image", e);
```
кастомный Exception

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/ImageForAdServiceImpl.java:
```diff
@@ -1,32 +1,147 @@
+            BufferedImage image = ImageIO.read(bis);
+
+            if (image == null) {
+                throw new IOException("Could not read image from file: " + filePath);
```
кастомный Exception

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/ImageForAdServiceImpl.java:
```diff
@@ -1,32 +1,147 @@
+            BufferedImage preview = new BufferedImage(newWidth, newHeight, BufferedImage.TYPE_INT_RGB);
+            Graphics2D graphics = preview.createGraphics();
+            graphics.drawImage(image, 0, 0, newWidth, newHeight, null);
+            graphics.dispose();
```
Это нам точно ни к чему, можно просто читать картинку

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/ImageForAdServiceImpl.java:
```diff
@@ -1,32 +1,147 @@
+                ImageIO.write(resizedImage, "jpg", outputStream);
+                return outputStream.toByteArray();
+            }
+        }
```
этот метод нам не нужен, такого ТЗ не было

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/ImageForAdServiceImpl.java:
```diff
@@ -1,32 +1,147 @@
     @Override
     public String getExtension(String filename) {
-        return null;
+        if (filename == null || filename.lastIndexOf(".") == -1) {
```
private и убрать из интерфейса

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/service/impl/ImageForAdServiceImpl.java:
```diff
@@ -1,32 +1,147 @@
+        return filename.substring(filename.lastIndexOf(".") + 1);
+    }
+
+    public Ad convertToAd(ExtendedAdDTO adDTO) {
```
Это задача маппера

Fet2113 committed:
add поле image.id в таблицу чтобы не падал фронт, fix AdMapper, UserMapper добавлены инструкции для image корретной работы фронта/

Rafnes committed:
[in progress] корректировки по комментариям

Rafnes committed:
[in progress] корректировки по комментариям 2 - добавлен FailedToReadFileException  и AuthServiceImpl рефактор - удалены связи OneToOne

ibogomolova committed:
[in progress] корректировки по комментариям 3 - refactor: rename Permissions to RoleAuthority, added custom Exception in ImageForAdServiceImpl, removed unnecessary methods from ImageForAdServiceImpl.java,

ibogomolova committed:
fix: String image in dto (пробовала в прошлый раз и случайно оставила так)

ibogomolova committed:
Merge remote-tracking branch 'origin/afetisov' into bogomolova

ibogomolova committed:
fix: "No property 'adId' found for type 'Image'"

ibogomolova committed:
корректировки по комментариям 3 (#71)

- refactor: rename Permissions to RoleAuthority, added custom Exception
in ImageForAdServiceImpl, removed unnecessary methods from
ImageForAdServiceImpl.java,

AleksandrZhukovJava commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -5,17 +5,26 @@
 @Configuration
 public class WebSecurityConfig {
 
+    private final UserRepository userRepository;
+
+    public WebSecurityConfig(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
```
А зачем он тут?
